File: craft_hr/events/employee_advance.py



import frappe

def validate(self, method=None):
        if not hasattr(self, "_original_advance_account"):
            self._original_advance_account = self.advance_account

        if self.is_leave_salary:
            if self.company:
                leave_account = frappe.db.get_value(
                    "Company", self.company, "leave_salary_advance_account"
                )
                if leave_account:
                    self.advance_account = leave_account
                else:
                    frappe.throw(f"No Leave Salary Advance Account set in Company {self.company}")
        else:
            if hasattr(self, "_original_advance_account"):
                self.advance_account = self._original_advance_account

File: craft_hr/public/js/employee_advance.js


frappe.ui.form.on("Employee Advance", {
    is_leave_salary: function(frm) {
        if (frm.doc.is_leave_salary) {
            frm.__original_advance_account = frm.doc.advance_account;

            if (frm.doc.company) {
                frappe.db.get_value("Company", frm.doc.company, "leave_salary_advance_account", (r) => {
                    if (r && r.leave_salary_advance_account) {
                        frm.set_value("advance_account", r.leave_salary_advance_account);
                    } else {
                        frappe.msgprint(__('No Leave Salary Advance Account set in Company {0}', [frm.doc.company]));
                    }
                });
            }
        } else {
            if (frm.__original_advance_account) {
                frm.set_value("advance_account", frm.__original_advance_account);
            }
        }
    }
});